lib/cereal/grpc/grpc.go

```diff
@@ -350,22 +347,24 @@ func (g *GrpcBackend) DequeueTask(ctx context.Context, taskName string) (*backen
 	taskCtx, cancel := context.WithCancel(ctx)
 	doneChan := make(chan error, 1)
 	go func() {
-		for {
-			msg, err := s.Recv()
-			if err != nil {
-				logrus.WithError(err).Debug("received error: canceling task context")
-				cancel()
-				doneChan <- err
-				return
-			}
-			if c := msg.GetCancel(); c != nil {
-				logrus.Debug("received cancel: canceling task context")
-				cancel()
-				doneChan <- errors.New("canceled")
-				return
-			}
+		// This goroutine will read the next, and what must be the last
+		// message.
+		var errOut error
+		msg, err := s.Recv()
+		if err != nil {
+			logrus.WithError(err).Debug("received error: canceling task context")
```

Comment:
I'm a little confused by these debug messages because we always cancel the task context at line 366 regardless of what case we hit.

Reply:
fair enough. there was a time when this wasn't true, ill fix up the messages